### DISCUSSION TITLE
permission to download data

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#7980] [Archive]
I have one question (I am still learning about directus capabilities) Why the user can still download all data to "csv" since he only has permission to view data and nothing else. He just has to see them and nothing else. Right side panel can't be modified? For example, only the administrator can download data...
Ok by the way, second question: Why is it not possible to download data in the form of "xls"? or it would be more interesting to create a report in some graphic form and display the data from directus in it...

By the way great job!!!!

**Comment by SeanDylanGoff** [comment]
Downloading is the same as viewing from a permissions standpoint - whether the data ends up in the browser or on disk shouldn't matter. A knowledgeable user could always do that even if there was a way to disable downloads.

CSV is a standard format that can be opened by virtually every program that's meant to view/edit tabular data. Including Excel... Supporting a proprietary format doesn't make sense in my opinion if we have a widely supported alternative.

disclaimer: i'm not part of directus

**Reply by koksikus** [reply]
Hi, thank you for answer.
Viewing the data only in the browser in my opinion protects against data theft (you have to rewrite several thousand records which is pointless) but downloading the entire database with one click makes it possible to further process whit this data. I might be wrong.

about csv - I can see how people cannot cope with importing data from csv on a daily basis. This is just for curiosity...
but an important solution would be to prepare a file something like a report, something that looks good "when printed"

**Reply by rijkvanzanten** [reply]
> Viewing the data only in the browser in my opinion protects against data theft 

Keep in mind that the data shown in Directus comes straight from the API in the same format, so disabling export in the app won't necessarily solve anything in regards to data theft. Besides, any tech savvy user would be able to easily scrape that data, even if it's only shown in the browser. (TL;DR, if it's visible in the browser, there's a way to scrape it).

As for the format, I'd have to agree with @SeanDylanGoff here. We should export to files that can be used in a wide range of programs, instead of limiting it to specific formats (if we were to support .xls, we should also support .numbers, and I'd rather not get into any of these proprietary formats if we don't have to)

**Reply by koksikus** [reply]
Thank you very much for your answer. 
I agree with you about scrape it and csv files, but all the time I mean users (who can only turn on the computer, open the file, see something and close the file and the computer) and they won't be able to open such a file (User friendly) and the other hands will not cope with scrape it.
The above is of little importance. More important, I have a topic that I have already mentioned to generate an invoice pdf or report and automatically enter data in the right place, as I found [in this solution](https://www.codespeedy.com/generate-pdf-invoice-using-java/). 
Second, if I would like to generate QR codes and read them later, how to enter, for example, such codes into directus [like this](https://www.codespeedy.com/generate-qr-code-of-text-and-url-in-javascript/) or [like this](https://elbruno.com/2014/02/20/html5-howto-scan-a-qrcode-in-a-webpage-using-javascript/).
Super mega great job with directus, I missed something like this for years, now it,s posible to manage the database - even movie library etc.
Will this types of extensions appear in the market? if yes then when?
Thank you

**Reply by koksikus** [reply]
Hi, I have one more question. Is it possible to enter an instruction like: if "the value of column" > "in a different column" and "the value of another column" = "true", then "enter the value in a specific column or in the one in which the instruction is located". Where can I find a hint? 
Thank you

**Reply by rijkvanzanten** [reply]
> Will this types of extensions appear in the market?

Potentially! It totally depends on the community. 

>  Is it possible to enter an instruction like: if "the value of column" > "in a different column" and "the value of another column" = "true", then "enter the value in a specific column or in the one in which the instruction is located". Where can I find a hint?

I'm not sure what you mean by that last "enter the value" bit, but I think what you're looking for is conditional fields https://docs.directus.io/guides/fields/#conditions

**Reply by rijkvanzanten** [reply]
All the tables and the relations are possible without a problem. What you describe as "Table 4" is called a "view" and is not currently supported in Directus (see https://github.com/directus/directus/discussions/3097 for the feature request)

**Comment by koksikus** [comment]
Anyone, Anything :)
Maybe I didn't understand what this software is for :)
I would like to learn something and I believe that Directus offers great opportunities.
